simulator/mips/mips_instr.h

```diff
@@ -89,39 +121,59 @@ class BaseMIPSInstr
         Trap trap = Trap::NO_TRAP;
 
         // Endian specific
-        const union _instr
+        const struct _instr
```

Comment:
That should go to the same header as constants. Moreover, we do not have to keep the object inside MIPSInstr class, and pass it between ctor and its nested functions.